docs/okeps/okep-5259-no-overlay.md

```diff
@@ -0,0 +1,882 @@
+  config: |
+    {
+            "cniVersion": "0.3.1",
+            "name": "l3-network",
+            "type": "ovn-k8s-cni-overlay",
+            "topology":"layer3",
+            "mtu": 1500,
+            "netAttachDefName": "default/l3-network",
+            "role": "primary",
+            "encapsulation": "none"
+    }
+```
+
+##### Gateway Router
+
+Currently, BGP routes to other node subnets are not imported into the gateway
+router, which ensures node-to-pod traffic traverses the OVN logical topology.
+However, we need the routes to be imported into the gateway router in no-overlay
+mode. We need to update the route import controller to check whether a network
+is in no-overlay mode and handle the route importing accordingly.
+
+##### Host Routing Tables
+
+To [maintain isolation between UDNs](https://github.com/ovn-kubernetes/ovn-kubernetes/pull/5140),
+node BGP speakers do not accept routes to NodeSubnets by default. However, to
+route east-west traffic in no-overlay mode, these routes must be imported to the
+host routing table. We need to update the node FRRConfiguration generation logic
+to check whether a network is in no-overlay mode and generate the node
+FRRConfiguration accordingly.
+
+#### No SNAT for egress traffic
+
+The OVN-Kubernetes BGP feature currently disables SNAT only for traffic destined
+to external networks, not for inter-pod communication across nodes.
+
+In no-overlay mode, the goal is to fully leverage BGP and eliminate the SNAT
+step on the gateway router, even for intra-cluster traffic. Within the same
+network, pod-to-pod traffic will utilize BGP-learned routes to reach its
+destination node directly, bypassing Geneve encapsulation and SNAT to the
+nodeIP.
+
+The only exception is for traffic destined to other nodes in the cluster
+(pod-to-other_node), to ensure nodePort services can be accessed across
+networks. This egress traffic will remain SNATed to the node IP before leaving
+the node. This is the current behavior when a network is advertised. We will not
+change it for no-overlay mode.
+
+#### Active traffic blocking between UDNs
+
+The [existing UDN isolation
+mechanism](https://github.com/ovn-kubernetes/ovn-kubernetes/pull/5186/commits/8f6e7d30ee5f4926a21e2de75488aad80344814b)
+will still be in place. Pods in different UDNs will be isolated from each other
+and the default network, utilizing the existing OVN ACLs on every node logical
+switch: ACLs are enforced on outgoing traffic at each advertised UDN switch,
+verifying whether both source and destination IPs belong to the same advertised
+UDN. If the destination IP does not come from the same UDN subnet as the source
+IP, traffic is dropped. No-overlay mode will not affect the isolation between
+UDNs or between UDNs and the default network. So, even if multiple UDNs were
+advertised to the same VRF, the cross-UDN traffic is still blocked. This is the
+current behavior when a network is advertised. We will not change it for no
+overlay mode.
+
```

Comment:
In the implementation section please flesh out what are the changes coming into:

1. cluster-manager
2. ovnkube-controller
3. ovnkube-node

components in detail - so either you can call them out in the above pieces or write them separately